### ISSUE TITLE
Configurable list of strings which are allowed to be duplicated

### ISSUE THREAD

**Issue by ferki** [gugod/Perl-Critic-TooMuchCode#25]
## TL;DR

As a developer, I would like to configure a list of string literals which are allowed to be duplicated, so I can avoid having to disable the TooMuchCode::ProhibitDuplicateLiteral policy either globally or via `## no critic ...` comments.

## Long version

Currently Perl::Critic::Policy::TooMuchCode::ProhibitDuplicateLiteral prohibits both strings and numbers to be duplicated, but it only allows configuring numbers to be excluded from the policy (via `whitelist_numbers`).

I often work with code (especially around [Rex](https://www.rexify.org)) where it's OK to have some strings duplicated, for example:

```perl
file '/path/to/directory', ensure => 'present';
file '/path/to/file',      ensure => 'present';
```

The workarounds for this I'm aware of are either disabling this policy completely, or sprinkle the code with `## no critic ...` comments. I would like to configure a list of `allowed_strings` instead.

I might come up with a PR for this soon, but anyone may feel free to implement it before me :)

**Comment by gugod** [comment]
This is now supported in 0.17 via "whitelist" parameter, which is based on pr #26 . Thanks @ferki !

**Comment by oalders** [comment]
I'm a bit late to the party here and I see that this has already shipped. I just wanted to note that something like `allowlist` is a good alternative to `whitelist`, since it conveys the same meaning without having to play off of white vs black. So, I think the naming choice in #26 clearly reflected what was trying to be achieved. Anyway, just something to think about. Thanks for all of your work on these tools. 😄 

There's some helpful info here: https://developers.google.com/style/word-list#blacklist I haven't gone over the whole guide myself, but what I have read has been quite helpful to me, so I thought I'd pass it along.

**Comment by ferki** [comment]
@oalders: thanks for the note, and you're right! On #26, I consciously chose the name `allowed_strings` for the new config option for this reason. I did not expect merging the functionality of the two variables (this fast at least :) ). So FTR, I would also support converging towards `allowlist` over `whitelist`.

Perhaps this kind of renaming is a good theme for a pull request?

**Comment by gugod** [comment]
I, for certain, supports the name change :). I will be grateful if anyone is willing to make PRs.

(evidently my job as the author is to make stupid mistakes as fast as possible.)


於 2021年9月21日 上午4:56:51 [GMT+09:00]，Ferenc Erki ***@***.***> 寫到：
***@***.***: thanks for the note, and you're right! On #26, I consciously chose the name `allowed_strings` for the new config option for this reason. I did not expect merging the functionality of the two variables (this fast at least :) ). So FTR, I would also support converging towards `allowlist` over `whitelist`.
>
>Perhaps this kind of renaming is a good theme for a pull request?
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/gugod/Perl-Critic-TooMuchCode/issues/25#issuecomment-923248812
--
Kang-min Liu

**Comment by gugod** [comment]
https://github.com/gugod/Perl-Critic-TooMuchCode/pull/28

I made this PR on my own. Feel free to double check.

It'll probably be release next week or at the beginnig of October. :)

**Comment by ferki** [comment]
Thank you, @gugod, for the follow-up changes and release! :rocket:
